oktapam/client/cloud_connection.go

```diff
@@ -0,0 +1,199 @@
+	}
+	if c.CloudConnectionDetails != nil {
+		flattenedDetails := make([]any, 1)
+		flattenedDetail := make(map[string]any)
+		if c.CloudConnectionDetails.AccountId != nil {
+			flattenedDetail[attributes.CloudConnectionAccountId] = *c.CloudConnectionDetails.AccountId
+		}
+		if c.CloudConnectionDetails.ExternalId != nil {
+			flattenedDetail[attributes.CloudConnectionExternalId] = *c.CloudConnectionDetails.ExternalId
+		}
+		if c.CloudConnectionDetails.RoleArn != nil {
+			flattenedDetail[attributes.CloudConnectionRoleARN] = *c.CloudConnectionDetails.RoleArn
+		}
+		flattenedDetails[0] = flattenedDetail
+		m[attributes.CloudConnectionDetails] = flattenedDetails
+	}
+
+	return m
+}
+
+func validateCloudConnectionData(cloudConnection CloudConnection) bool {
+	nameRegex, nameRegexErr := regexp.Compile(`^[A-Za-z0-9-_.]+$`)
+	if nameRegexErr != nil {
+		fmt.Println("invalid cloud connection name")
+		return false
+	}
+	nameValidation := nameRegex.MatchString(*cloudConnection.Name) && len(*cloudConnection.Name) > 1
+
+	accountIdRegex, accountIdRegexErr := regexp.Compile(`^\d{12}$`)
+	if accountIdRegexErr != nil {
+		fmt.Println("invalid cloud connection account id")
+		return false
+	}
+	accountIdValidation := accountIdRegex.MatchString(*cloudConnection.CloudConnectionDetails.AccountId)
+
+	providerValidation := *cloudConnection.Provider == "aws"
+	externalIdValidation := len(*cloudConnection.CloudConnectionDetails.ExternalId) != 0
+	roleArnValidation := len(*cloudConnection.CloudConnectionDetails.RoleArn) != 0
+
+	return nameValidation && accountIdValidation && providerValidation && externalIdValidation && roleArnValidation
+}
+
+func (c OktaPAMClient) ListCloudConnections(ctx context.Context) ([]CloudConnection, error) {
+	requestURL := fmt.Sprintf("/v1/teams/%s/cloud_connections", url.PathEscape(c.Team))
+	cloudConnections := make([]CloudConnection, 0)
+
+	for {
+		// List will paginate, so we make a request, add results to array to return, check if we get a next page, and if so loop again
+		logging.Tracef("making GET request to %s", requestURL)
+
+		resp, err := c.CreateBaseRequest(ctx).SetResult(&CloudConnectionsListResponse{}).Get(requestURL)
+		if err != nil {
+			logging.Errorf("received error while making request to %s", requestURL)
+			return nil, err
+		}
+		if _, err := checkStatusCode(resp, http.StatusOK); err != nil {
+			return nil, err
+		}
+
+		cloudConnectionsListResponse := resp.Result().(*CloudConnectionsListResponse)
+		cloudConnections = append(cloudConnections, cloudConnectionsListResponse.CloudConnections...)
+
+		linkHeader := resp.Header().Get("Link")
+		if linkHeader == "" {
+			break
+		}
+		links := linkheader.Parse(linkHeader)
+		requestURL = ""
+
+		for _, link := range links {
+			if link.Rel == "next" {
+				requestURL = link.URL
+				break
+			}
+		}
+	}
+
+	return cloudConnections, nil
+}
+
+func (c OktaPAMClient) GetCloudConnection(ctx context.Context, id string) (*CloudConnection, error) {
```

Comment:
nit:  lets name the vars more freely!
```suggestion
func (c OktaPAMClient) GetCloudConnection(ctx context.Context, cloudConnectionId string) (*CloudConnection, error) {
```

Reply:
Done.